frame/support/src/lib.rs

```diff
@@ -1550,17 +1554,17 @@ pub mod pallet_prelude {
 ///
 /// ```
 /// #[frame_support::pallet]
-/// // NOTE: Example is name of the pallet, it will be used as unique identifier for storage
+/// // NOTE: The name of the pallet will be provided by construct_runtime, and will be used as
+/// // unique identifier for storage. It is not defined in the pallet itself.
 /// pub mod pallet {
 /// 	use frame_support::pallet_prelude::*; // Import various types used in pallet definition
-/// 	use frame_system::pallet_prelude::*; // OriginFor helper type for implementing dispatchables.
+/// 	use frame_system::pallet_prelude::*; // Import some system helper types.
 ///
 /// 	type BalanceOf<T> = <T as Config>::Balance;
 ///
 /// 	// Define the generic parameter of the pallet
-/// 	// The macro checks trait generics: is expected none or `I = ()`.
-/// 	// The macro parses `#[pallet::constant]` attributes: used to generate constant metadata,
-/// 	// expected syntax is `type $IDENT: Get<$TYPE>;`.
+/// 	// The macro parses `#[pallet::constant]` attributes, and use them to generate constant
+/// 	// metadata,
```

Comment:
```suggestion
/// 	// The macro parses `#[pallet::constant]` attributes and uses them to generate
/// 	// metadata for the pallet's constants.
```

Reply:
done in https://github.com/paritytech/substrate/pull/7922/commits/1255db1d731af7e07f74bbeaf9a30009d4e06438